src/schemes/boundary/open_boundary/boundary_zones.jl

```diff
@@ -0,0 +1,336 @@
+@doc raw"""
+    InFlow(; plane, flow_direction, density, particle_spacing,
+           initial_condition=nothing, extrude_geometry=nothing,
+           open_boundary_layers::Integer)
+
+Inflow boundary zone for [`OpenBoundarySPHSystem`](@ref).
+
+The specified plane (line in 2D or rectangle in 3D) will be extruded in upstream
+direction (the direction opposite to `flow_direction`) to create a box for the boundary zone.
+There are three ways to specify the actual shape of the inflow:
+1. Don't pass `initial_condition` or `extrude_geometry`. The boundary zone box will then
+   be filled with inflow particles.
+2. Pass a 1D shape in 2D or a 2D shape in 3D, which is then extruded in upstream direction
+   to create the inflow particles.
+   In 2D, the shape must be either an initial condition with 2D coordinates, which lies
+   on the line specified by `plane`, or an initial condition with 1D coordinates, which lies
+   on the line specified by `plane` when a y-coordinate of `0` is added.
+   In 3D, the shape must be either an initial condition with 3D coordinates, which lies
+   in the rectangle specified by `plane`, or an initial condition with 2D coordinates,
+   which lies in the rectangle specified by `plane` when a z-coordinate of `0` is added.
+   Note that particles outside the boundary zone box will be removed.
+3. Pass a 2D initial condition in 2D or a 3D initial condition in 3D, which will be used
```

Comment:
```suggestion
3. Pass an initial condition, which will be used
```

Reply:
Here, the information if it's 2D or 3D is important, IMO (see second point).

Reply:
So 2. is passing both initial_condition and extrude_geometry?

Reply:
1. `initial_condition=nothing` and `extrude_geometry=nothing` (default)
2. specify  `extrude_geometry` (ignore kwarg `initial_condition`)
3. specify `initial_condition` (ignore kwarg `extrude_geometry`)

Reply:
Ok than start the subpoints with exactly that information. Especially the text under 2 reads like you need to have both.

Reply:
good point. thx